primitives/core/src/lib.rs

```diff
@@ -389,6 +389,45 @@ macro_rules! impl_maybe_marker {
 	}
 }
 
+/// Macro for creating `Maybe*` marker traits.
+///
+/// Such a maybe-marker trait requires the given bound when `feature = std` or `feature =
+/// serde` but does not require the bound in `no_std` without `serde` activated.
+///
+/// # Example
+///
+/// ```
+/// sp_core::impl_maybe_marker_serde! {
+///     /// A marker for a type that implements `Debug` when `feature = serde` or `feature = std`.
+///     trait MaybeDebug: std::fmt::Debug;
+///     /// A marker for a type that implements `Debug + Display` when `feature = serde` or `feature = std`.
+///     trait MaybeDebugDisplay: std::fmt::Debug, std::fmt::Display;
+/// }
+/// ```
+#[macro_export]
+macro_rules! impl_maybe_marker_serde {
```

Comment:
```suggestion
macro_rules! impl_maybe_marker_std_or_serde {
```

Make the name for explicit.